scopesim/effects/detector_list.py

```diff
@@ -144,16 +145,20 @@ def detector_headers(self, ids=None):
     def plot(self):
         import matplotlib.pyplot as plt
 
+        fig = plt.gcf()
```

Comment:
Idea: I think the Pythonic way to plot something in a `plot` function is to give the axis where the plot should go on as an argument to the `plot` function. I'm struggling a bit with how to do this properly myself, so this is just sharing some thoughts.